core/actor/src/main/scala/net/liftweb/actor/LAFuture.scala

```diff
@@ -106,19 +106,31 @@ class LAFuture[T](val scheduler: LAScheduler) {
    */
   def map[A](f: T => A): LAFuture[A] = {
     val ret = new LAFuture[A](scheduler)
-    onComplete(v => ret.complete(v.map(f)))
+    onComplete(v => ret.complete(v.flatMap(wrapWithBox(f))))
     ret
   }
 
   def flatMap[A](f: T => LAFuture[A]): LAFuture[A] = {
     val ret = new LAFuture[A](scheduler)
     onComplete(v => v match {
-      case Full(v) => f(v).onComplete(v2 => ret.complete(v2))
+      case Full(v) =>
+        wrapWithBox(f)(v) match {
+          case Full(successfullyComputedFuture) => successfullyComputedFuture.onComplete(v2 => ret.complete(v2))
+          case e: EmptyBox => ret.complete(e)
+        }
       case e: EmptyBox => ret.complete(e)
     })
     ret
   }
 
+  private def wrapWithBox[A](f: T => A)(t: T): Box[A] = {
```

Comment:
It's allmost the same but there is a little differences: it takes a function not a call-by-name parameter. In tryo you must write:
```scala
box.flatMap { v =>
  tryo(f(v))
}
```

instead of:
```scala
box.flatMap(wrapWithBox(f))
```

The second thing is that module *util* uses module *actor* - at least in *Mailer* there is used *SpecializedLiftActor*. But we can solve this, moving tryo into module *common* (which is IMO the better place for it).